src/playlist-controller.js

```diff
@@ -2052,4 +2099,128 @@ export class PlaylistController extends videojs.EventTarget {
       videoDuration
     });
   }
+
+  pathwayAttribute_(playlist) {
+    return playlist.attributes['PATHWAY-ID'] || playlist.attributes.serviceLocation;
+  }
+  /**
+   * Initialize content steering listeners and apply the tag properties.
+   */
+  initContentSteeringController_() {
+    const initialMain = this.main();
+
+    if (!initialMain.contentSteering) {
+      return;
+    }
+
+    const updateSteeringValues = (main) => {
+      for (const playlist of main.playlists) {
+        this.contentSteeringController_.addAvailablePathway(this.pathwayAttribute_(playlist));
+      }
+
+      this.contentSteeringController_.assignTagProperties(main.uri, main.contentSteering);
+    };
+
+    updateSteeringValues(initialMain);
+
+    this.contentSteeringController_.on('content-steering', this.excludeThenChangePathway_.bind(this));
+
+    // We need to ensure we update the content steering values when a new
+    // manifest is loaded in live DASH with content steering.
+    if (this.sourceType_ === 'dash') {
+      this.mainPlaylistLoader_.on('mediaupdatetimeout', () => {
+        this.mainPlaylistLoader_.refreshMedia_(this.mainPlaylistLoader_.media().id);
+
+        // clear past values
+        this.contentSteeringController_.abort();
+        this.contentSteeringController_.clearTTLTimeout_();
+        this.contentSteeringController_.availablePathways_.clear();
```

Comment:
we should provide a public interface to work with private stuff

Reply:
Added this public function here, let me know if that is what you were thinking: https://github.com/videojs/http-streaming/pull/1427/commits/e882de56db6a14be2e74b41ccd16ceba0e67cccf#diff-a167ea6828d7c01e63609400b5cb67b841a64432c629f7913fc04bc528fa3f3cR420